tools/automation.py

```diff
@@ -229,20 +229,64 @@ def xlsx_to_yaml_parser(mentor_row, mentor_index):
     return mentor
 
 
-def get_all_mentors_in_yml_format(xlsx_file_path, skip_rows=0):
+def get_yml_data(yml_file_path):
     """
-    Read all mentors from Excel sheet.
-    Prepare data for writing to yaml file.
+    Get data from mentors.yml.
+    Return dataframe with name, index, sort and disabled values.
     """
-    # list of dict
-    mentors = []
+    yml_dict = read_yml_file(yml_file_path)
+
+    logging.info(f"Mentors yml total: {len(yml_dict)}")
+
+    yml_data = []
+
+    for mentor in yml_dict:
+        yml_data.append([mentor['name'].lower(),
+                        mentor['index'],
+                        mentor['disabled'],
+                        mentor['sort']])
+
+    df_yml_data = pd.DataFrame(yml_data,
+                            columns=['Name', 'Index', 'Disabled', 'Sort'])
+    return df_yml_data
 
+
+def get_all_mentors_in_yml_format(yml_file_path, xlsx_file_path, skip_rows=0):
+    """
+    Read all mentors from Excel sheet:
+     - if mentor is in current mentors.yml, use existing values for index, sort and disabled.
+     - if mentor is new, continue indexing from the largest index from current mentors.yml
+    """
     df_mentors = pd.read_excel(xlsx_file_path, sheet_name=SHEET_NAME, skiprows=skip_rows)
 
     logging.info(f"Excel read {len(df_mentors)} mentors")
 
+    # Get current mentors' data (name, index, disabled, sort) from mentors.yml
+    df_yml = get_yml_data(yml_file_path)
+
+    # Calculate index for new mentors
+    new_index = 1
+    if not df_yml.empty:
+        new_index = df_yml['Index'].max().item() + 1
+
+    # list of dict
+    mentors = []
+
     for row in range(0, len(df_mentors)):
-        mentor = xlsx_to_yaml_parser(df_mentors.iloc[row], row + 1)
+        mentor_name = df_mentors.iloc[row].values[1].lower()
+
+        df_yml_row = df_yml.loc[df_yml.Name == mentor_name]
```

Comment:
Using dataframe since it is very easy to check if the mentor_name from xlsx table is among the mentors in mentors.yml